src/ua_connection.c

```diff
@@ -167,6 +171,7 @@ UA_Connection_processChunks(UA_Connection *connection, void *application,
     do {
         retval = processChunk(connection, application, processCallback,
                               &pos, end, &done);
+        connection->pendingMessage = false;
```

Comment:
Here we should test if the current chunk is at the end of the received buffer.
Otherwise the second chunk always has pendingMessage = false.

What if there are three chunks in one TCP packet?

Reply:
After each processChunk pendingMessage is set to false. If we have three chunks,  pendingMessage is set multiple time to true.

In fact processChunk only process one message.